teps/0044-data-locality-and-pod-overhead-in-pipelines.md

```diff
@@ -301,11 +301,78 @@ controller to make this decision.
 
 ## Design proposal
 
-TBD - currently focusing on enumerating and examining alternatives before selecting one or more ways forward.
+### Summary
+
+The Tekton Pipelines controller will be updated to support running a Pipeline in a pod.
+Users will choose which parts of a pipeline to run in a pod by grouping them into a sub-Pipeline, using the [Pipelines in Pipelines](./0056-pipelines-in-pipelines.md) feature.
+Any Pipelines grouped under a Pipeline executed in a pod will also execute in that pod.
+
+We will maintain the 1-1 relationship between Tasks and TaskRuns by creating one TaskRun per Pipeline Task and using it as the source of truth for the status of the
+execution of that Task.
+
+Additional design details and naming still under discussion.
+
+### Rollout plan
+
+#### Required Features
+- **Sequential Tasks**. This TEP aims to mitigate the challenges of passing data between subsequent Tasks, such as a `git clone` followed by a `docker build`, so we must
+support sequential Tasks in one pod.
+- **Workspaces and Results**. These features are also critical for passing data between Tasks.
+- **Finally Tasks**. Finally Tasks should be supported for use cases which involve uploading or processing PipelineRun outputs,
+such as data from previous Tasks contained in Workspaces.
+- **Parallel Tasks**. Another primary use case for this TEP is to support parallel Tasks that share data, which today typically must be run in separate pods on the same node.
+Running parallel Tasks on the same pod wouldn't allow computation to be balanced between nodes, but it would allow users to avoid paying performance and UX penalties
+associated with PVCs or other workspace backing options. In addition, `finally` Tasks, which are also a required feature, run in parallel.
+- **Params**. Params are critical to the reusability of many catalog Tasks, and implementation in a pod will require few, if any, changes.
+
+#### Optional Features
+
+The following features are useful for PipelineRuns executed in pods for the same reasons that they are useful in PipelineRuns
+executed in multiple pods. Not all of these features will be equally easy to implement in a pod.
+
+- **`when` expressions**
+- **retries**
+- **timeouts**
+- **task-level sidecars**
+- **Matrixed Tasks (fanned out from params)**
+
```

Comment:
could you please add some rationale on why these features are optional? also, can we reframe this as features that will be supported in future work instead of optional?